### ISSUE TITLE
Monitor processing of uploads

### ISSUE THREAD

**Issue by willwhite** [mapbox/mapbox-studio-classic#1120]
There are two phases to uploading a file to Mapbox:
1. Uploading the file to our servers
2. Waiting for the servers to process the file

Right now, Mapbox Studio treats an upload as complete after step 1. However, the upload isn't really "complete" until after step 2.

This is confusing from a user's perspective, because Mapbox Studio will claim an upload is "finished" but the upload won't appear on the styles or data page until it is processed. Processing can take significant time. Processing can also fail and the user has no way of know.

Solution: After finishing step 1, Mapbox Studio should monitor the processing phase and provide the user with feedback if the processing fails.

Much of this functionality can be added to mapbox-upload.

/cc @camilleanne @samanpwbb @geografa

**Comment by camilleanne** [comment]
I saw this one coming -- there are a couple of solutions that I can see off the top of my head:
1) add a bit of ui fairy dust to the export interface that once the upload is complete and the task has hit 100% a spinning "processing upload" screen is shown until the second response gets back to Mapbox Studio.
2) Can the upload actually be canceled still at the processing stage? Should the task end when the upload has reached 100% and we let the server take care of processing it on it's own?
3) (least likely) Figure out a way to estimate how long it will take to process the file and send that down.

**Comment by samanpwbb** [comment]
Mapbox Studio could listen for changes to the user's upload listing and report back when something changes. This would be useful not just for reporting failures but also for reporting when an uploaded source is ready to be used.

**Comment by willwhite** [comment]
Yeah, per chat with @samanpwbb we are focusing on making the workflow smoother between Studio and Mapbox.com first. Then we can see if this kind of tighter integration is needed.

**Comment by cossatot** [comment]
It'd be really helpful if Mapbox Studio were able to figure out if the max tile size is exceeded before commencing with the upload, and maybe offering a warning or a (default) option to cancel.  If the data is going to be rejected, why go ahead with the upload, when that can take minutes to tens of minutes for large datasets?

**Comment by samanpwbb** [comment]
@cossatot ~~I have this started in https://github.com/mapbox/mapbox-studio/pull/1022 - will get it merged soon :)~~ Just merged, so it'll be in the next release

**Comment by cossatot** [comment]
@samanpwbb Awesome, looking forward to it.
